x-pack/plugin/ent-search/src/test/java/org/elasticsearch/xpack/application/connector/ConnectorSyncInfoTests.java

```diff
@@ -0,0 +1,50 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+package org.elasticsearch.xpack.application.connector;
+
+import org.elasticsearch.common.io.stream.NamedWriteableRegistry;
+import org.elasticsearch.common.settings.Settings;
+import org.elasticsearch.search.SearchModule;
+import org.elasticsearch.test.ESTestCase;
+import org.junit.Before;
+
+import java.io.IOException;
+import java.util.List;
+
+import static java.util.Collections.emptyList;
+import static org.hamcrest.CoreMatchers.equalTo;
+
+public class ConnectorSyncInfoTests extends ESTestCase {
```

Comment:
I see you've done `AbstractBWCSerializationTestCase`s later on. Great work! 👏 

Are these individual tests used for the classes that are used for the internal index representation and not included into request / responses? If so, you did double great work 👏 👏

Reply:
The internal classes for now are only used only for internal index representation and are not included in request/response. In the next PRs:
- `Connector` would be returned from `GET` request (to be implemented)
- Some classes, that were used to represent connector properties, will overlap with our desired API schema (request) - in that case I will just implement parsers there and will use them for parsing request JSON. In that case I will add thorough parsing <-> toXContnet tests to see if representation works fine :)

Reply:
Thanks for clarifying! 👍